src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,76 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+gchar*
+read_hwmon_file (gchar *path, GError **error)
+{
+  gchar *content = NULL;
+
+  if (!g_file_get_contents (path, &content, NULL, error))
+    {
+      if (!g_error_matches (*error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_message ("error loading name: %s", (*error)->message);
+    }
+
+  return content;
+}
+
+void
+sample_cpu_sensors (CockpitSamples *samples, int hwmonID)
+{
+  for (int i = 1; TRUE; i++)
+    {
+      g_autoptr(GError) error = NULL;
+      g_autofree gchar *temp = g_strdup_printf ("/sys/class/hwmon/hwmon%d/temp%d_input", hwmonID, i);
+      g_autofree gchar *temp_content = read_hwmon_file (temp, &error);
+
+      if (error != NULL)
```

Comment:
The way this is currently written, this will match `ENOENT` as well. See monster comment above for some options.